kernel/src/engine/arrow_footer_skipping.rs

```diff
@@ -0,0 +1,359 @@
+        };
+        Some(value)
+    }
+
+    fn get_max_stat_value(&self, col: &ColumnPath, data_type: &DataType) -> Option<Scalar> {
+        use PrimitiveType::*;
+        let value = match (data_type.as_primitive_opt()?, self.get_stats(col)?) {
+            (String, Statistics::ByteArray(s)) => s.max_opt()?.as_utf8().ok()?.into(),
+            (String, Statistics::FixedLenByteArray(s)) => s.max_opt()?.as_utf8().ok()?.into(),
+            (String, _) => None?,
+            (Long, Statistics::Int64(s)) => s.max_opt()?.into(),
+            (Long, Statistics::Int32(s)) => (*s.max_opt()? as i64).into(),
+            (Long, _) => None?,
+            (Integer, Statistics::Int32(s)) => s.max_opt()?.into(),
+            (Integer, _) => None?,
+            (Short, Statistics::Int32(s)) => (*s.max_opt()? as i16).into(),
+            (Short, _) => None?,
+            (Byte, Statistics::Int32(s)) => (*s.max_opt()? as i8).into(),
+            (Byte, _) => None?,
+            (Float, Statistics::Float(s)) => s.max_opt()?.into(),
+            (Float, _) => None?,
+            (Double, Statistics::Double(s)) => s.max_opt()?.into(),
+            (Double, _) => None?,
+            (Boolean, Statistics::Boolean(s)) => s.max_opt()?.into(),
+            (Boolean, _) => None?,
+            (Binary, Statistics::ByteArray(s)) => s.max_opt()?.data().into(),
+            (Binary, Statistics::FixedLenByteArray(s)) => s.max_opt()?.data().into(),
+            (Binary, _) => None?,
+            (Date, Statistics::Int32(s)) => Scalar::Date(*s.max_opt()?),
+            (Date, _) => None?,
+            (Timestamp, Statistics::Int64(s)) => Scalar::Timestamp(*s.max_opt()?),
+            (Timestamp, _) => None?, // TODO: Int96 timestamps
+            (TimestampNtz, Statistics::Int64(s)) => Scalar::TimestampNtz(*s.max_opt()?),
+            (TimestampNtz, _) => None?, // TODO: Int96 timestamps
+            (Decimal(..), _) => None?,  // TODO: Decimal (Int32, Int64, FixedLenByteArray)
+        };
+        Some(value)
+    }
+
+    fn get_nullcount_stat_value(&self, col: &ColumnPath) -> Option<i64> {
+        // Null stats always have the same type (u64), so we can handle them directly. Further,
+        // the rowcount stat is i64 so we can safely cast this to i64 to match
+        Some(self.get_stats(col)?.null_count_opt()? as i64)
+    }
+
+    fn get_rowcount_stat_value(&self) -> i64 {
+        self.row_group.num_rows()
+    }
+}
+
+pub(crate) fn partial_cmp_scalars(
+    a: &Scalar,
+    b: &Scalar,
+    ord: Ordering,
+    inverted: bool,
+) -> Option<bool> {
+    let result = a.partial_cmp(b)? == ord;
+    Some(result != inverted)
+}
+
+pub(crate) fn col_name_to_path(col: &str) -> ColumnPath {
+    // TODO: properly handle nested columns
+    // https://github.com/delta-incubator/delta-kernel-rs/issues/86
+    ColumnPath::new(col.split('.').map(|s| s.to_string()).collect())
+}
+
+pub(crate) fn compute_field_indices(
+    fields: &[ColumnDescPtr],
```

Comment:
I have similar functionality in my [PR](https://github.com/delta-incubator/delta-kernel-rs/pull/349/files), but I just need to extract the columns (I don't have access to a `&[ColumnDescPtr]`). It may be a good idea to split out the `expression_to_column` part so I could reuse your implementation once you merge.